api/rest/swagger.yaml

```diff
@@ -0,0 +1,796 @@
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Peer'
+              example: {"id":"QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","addresses":["/p2p-circuit/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/127.0.0.1/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/192.168.86.35/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/10.6.0.2/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/192.168.1.42/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH"],"cluster_peers":["QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH"],"cluster_peers_addresses":null,"version":"0.10.0","commit":"","rpc_protocol_version":"/ipfscluster/0.10/rpc","error":"","ipfs":{"id":"QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","addresses":["/ip4/127.0.0.1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/192.168.86.35/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/10.0.3.1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/10.6.0.2/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/192.168.1.42/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip6/::1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse"],"error":""},"peername":"omou"}
+  /version:
+    get:
+      summary: Get Version of Cluster
+      tags:
+        - cluster
+      description: Get the current version of the IPFS Cluster.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  Version:
+                    type: string
+              example: {"Version":"0.10.0"}
+              
+  /add:
+    post:
+      summary: Add a File
+      parameters:
+        - $ref: '#/components/parameters/pinNameParam'
+        - $ref: '#/components/parameters/pinReplicationMinParam'
+        - $ref: '#/components/parameters/pinReplicationMaxParam'
+        - $ref: '#/components/parameters/pinShardSize'
+        - $ref: '#/components/parameters/pinMetaParam'
+        - $ref: '#/components/parameters/hash'
+        - $ref: '#/components/parameters/recursive'
+        - $ref: '#/components/parameters/layout'
+        - $ref: '#/components/parameters/rawLeaves'
+        - $ref: '#/components/parameters/hidden'
+        - $ref: '#/components/parameters/wrap'
+        - $ref: '#/components/parameters/shard'
+        - $ref: '#/components/parameters/progress'
+        - $ref: '#/components/parameters/cidVersion'
+        - $ref: '#/components/parameters/streamChannels'
+        - $ref: '#/components/parameters/noCopy'
+        - $ref: '#/components/parameters/chunker'
+      tags:
+        - add
+      description: Add a file to IPFS via IPFS Cluster
```

Comment:
`This endpoint adds a file to the allocated IPFS daemons and pins it in the cluster. The cluster peer receiving the request will chunk and build the DAG, forwarding the resulting blocks to their final destination.`